src/java/io/bazel/rulesscala/scalac/deps_tracking_reporter/after_2_13_12/DepsTrackingReporter.java

```diff
@@ -0,0 +1,314 @@
+                int strictDepsCount = 0;
+                int compilerDepsCount = 0;
+                for (Dependency dep : usedDeps) {
+                    String depReport = addDepMessage(dep);
+                    if (dep.getIgnored()) {
+                        continue;
+                    }
+
+                    if (directTargets.contains(dep.getLabel())) {
+                        continue;
+                    }
+
+                    if (dep.getKind() == Kind.EXPLICIT) {
+                        strictDepsCount++;
+                        strictDepsReport
+                                .append(isWarning ? "warning: " : "error: ")
+                                .append(depReport);
+                    } else {
+                        compilerDepsCount++;
+                        compilerDepsReport
+                                .append(isWarning ? "warning: " : "error: ")
+                                .append(depReport);
+                    }
+                }
+
+                if (strictDepsCount > 0) {
+                    if (ops.strictDepsMode.equals("warn")) {
+                        reporter.warning(NoPosition$.MODULE$, strictDepsReport.toString(), List$.MODULE$.<CodeAction>empty());
+                    } else {
+                        reporter.error(NoPosition$.MODULE$, strictDepsReport.toString(), List$.MODULE$.<CodeAction>empty());
+                    }
+                }
+
+                if (!ops.compilerDepsMode.equals("off") && compilerDepsCount > 0) {
+                    if (ops.compilerDepsMode.equals("warn")) {
+                        reporter.warning(NoPosition$.MODULE$, compilerDepsReport.toString(), List$.MODULE$.<CodeAction>empty());
+                    } else {
+                        reporter.error(NoPosition$.MODULE$, compilerDepsReport.toString(), List$.MODULE$.<CodeAction>empty());
+                    }
+                }
+            }
+
+            if (!ops.unusedDependencyCheckerMode.equals("off")) {
+                boolean isWarning = ops.unusedDependencyCheckerMode.equals("warn");
+                StringBuilder unusedDepsReport = new StringBuilder("Unused dependencies:\n");
+                int count = 0;
+                for (Dependency dep : unusedDeps) {
+                    if (dep.getIgnored()) {
+                        continue;
+                    }
+                    count++;
+                    unusedDepsReport
+                            .append(isWarning ? "warning: " : "error: ")
+                            .append(removeDepMessage(dep));
+                }
+                if (count > 0) {
+                    if (isWarning) {
+                        reporter.warning(NoPosition$.MODULE$, unusedDepsReport.toString(), List$.MODULE$.<CodeAction>empty());
+                    } else if (ops.unusedDependencyCheckerMode.equals("error")) {
+                       //reporter.error(NoPosition$.MODULE$, unusedDepsReport.toString(), List$.MODULE$.<CodeAction>empty());
```

Comment:
So now it build fine by adding an empty Scala `List[Action]` to warning/error methods but somehow it triggers a behaviour change where now `bazel build //... --repo_env=SCALA_VERSION=2.13.12` within `test_version/test_scala_version-{timestamp}` directory  will exit and report unused dependencies...

```
DT:bazel-out/k8-fastbuild/bin/external/io_bazel_rules_scala_scala_library/io_bazel_rules_scala_scala_library.stamp/scala-library-2.13.12-stamped.jar
warning: 1 deprecation (since 2.13.0); re-run with -deprecation for details
1 warning
error: Unused dependencies:
error: Target '//:MixJavaScalaLib' (via jar: ' bazel-out/k8-fastbuild/bin/MixJavaScalaLib_java.jar ')  is specified as a dependency to //:MixJavaScalaLibBinary but isn't used, please remove it from the deps.
You can use the following buildozer command:
buildozer 'remove deps //:MixJavaScalaLib' //:MixJavaScalaLibBinary
error: Target '//:MixJavaScalaLib' (via jar: ' bazel-out/k8-fastbuild/bin/MixJavaScalaLib.jar ')  is specified as a dependency to //:MixJavaScalaLibBinary but isn't used, please remove it from the deps.
You can use the following buildozer command:
buildozer 'remove deps //:MixJavaScalaLib' //:MixJavaScalaLibBinary
Build failed
java.lang.RuntimeException: Build failed
	at io.bazel.rulesscala.scalac.ScalacWorker.compileScalaSources(ScalacWorker.java:324)
	at io.bazel.rulesscala.scalac.ScalacWorker.work(ScalacWorker.java:72)
	at io.bazel.rulesscala.worker.Worker.persistentWorkerMain(Worker.java:86)
	at io.bazel.rulesscala.worker.Worker.workerMain(Worker.java:39)
	at io.bazel.rulesscala.scalac.ScalacWorker.main(ScalacWorker.java:
```

Cannot even reproduce this behaviour with Scala  `2.13.11` and from what I can tell default is `unused_dependency_checker_mode = "off"` for both cases :thinking: Will continue to dig as it must be related to my changes...

Reply:
With Scala `2.13.12`, I think the `//third_party/dependency_analyzer` is now finding unused dependencies.